src/platform/Linux/DnssdImpl.cpp

```diff
@@ -850,11 +850,18 @@ CHIP_ERROR ChipDnssdFinalizeServiceUpdate()
 }
 
 CHIP_ERROR ChipDnssdBrowse(const char * type, DnssdServiceProtocol protocol, chip::Inet::IPAddressType addressType,
-                           chip::Inet::InterfaceId interface, DnssdBrowseCallback callback, void * context)
+                           chip::Inet::InterfaceId interface, DnssdBrowseCallback callback, void * context,
+                           intptr_t * browseIdentifier)
 {
+    *browseIdentifier = reinterpret_cast<intptr_t>(nullptr);
     return MdnsAvahi::GetInstance().Browse(type, protocol, addressType, interface, callback, context);
 }
 
+CHIP_ERROR ChipDnssdStopBrowse(intptr_t browseIdentifier)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
```

Comment:
It would be nice if we could come up with a unified solution.

Looking at avahi, it seems that the way it is usually aborted is by just stopping its polling loop.  Our `MdnsAvahi::Shutdown` does do this, so we have this rudimentary ability to get-no-more-callbacks from avahi, but that is obviously not granular enough here.

Is it possible for us to say what we would like the cancellation API to be and for this desired API to at least be theoretically implementable on all of the target platforms?  It feels odd to be introducing something that is definitely not implementable everywhere.

Reply:
Actually Tizen supports this workflow. There is an API function `dnssd_stop_browsing_service()`.

Some time ago I was trying to implement the same thing - stop mDNS browse. However, I got lost in the CHIPDeviceController internals... I was not sure who should initiate this stop action :/

Reply:
Good to know.  If several back ends can support this, it seems valuable.

That still leaves the difficulty of where and when to call it given it isn't available everywhere.  The answer in this PR is:

```
    mCommissioner->StopCommissionableDiscovery();
    return CHIP_NO_ERROR;
```

I.e., return value from the call isn't considered.  Given our current limitations, that is probably reasonable.

Reply:
I think for various backends we can in fact implement this API; I just do not know the backends enough to do it (quickly).

Some backends may not support "stop browse" in any sane way; e.g. the minimal dns-sd impl (which doesn't use this API at all, of course) has no concept of stopping a browse.

Fundamentally, this API is a way to notify the backend that "we are going to ignore results from this browse now; if you allocated resources to handle the browse, you should free those".  Some backends will not have any resources to deallocate.  Those could probably return CHIP_NO_ERROR instead of CHIP_ERROR_NOT_IMPLEMENTED...

Reply:
There is a slight nuance to this:

> if you allocated resources to handle the browse, you should free those

In e.g. avahi, I don't see that we would be able to cleanly free the avahi structures.  But the shim for avahi could be made to no longer use caller-provided resources with the result that the caller could be safe to move on and free anything it has associated with the browse.

But that is just an implementation detail.

So yes, agreed that we can do this with several of the platforms.  Also agreed that this is the right approach for return value:

>  Some backends will not have any resources to deallocate. Those could probably return CHIP_NO_ERROR instead of CHIP_ERROR_NOT_IMPLEMENTED...

Yes, CHIP_NO_ERROR should indicate that resources were freed / browse was stopped, or that there were no resources to free.

Reply:
Per Slack discussion, doing the various backends in followups is fine.